src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -69,26 +72,46 @@ def create_consent_request(
             "Application redis cache required, but it is currently disabled! Please update your application configuration to enable integration with a redis cache."
         )
 
-    if not data.email:
-        raise HTTPException(HTTP_400_BAD_REQUEST, detail="An email address is required")
-
-    identity = ProvidedIdentity.filter(
-        db=db,
-        conditions=(
-            (ProvidedIdentity.field_name == ProvidedIdentityType.email)
-            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(data.email))
-            & (ProvidedIdentity.privacy_request_id.is_(None))
-        ),
-    ).first()
+    if not data.email and not data.phone_number:
+        raise HTTPException(
+            HTTP_400_BAD_REQUEST,
+            detail="An email address or phone number identity is required",
+        )
 
-    if not identity:
-        provided_identity_data = {
-            "privacy_request_id": None,
-            "field_name": "email",
-            "hashed_value": ProvidedIdentity.hash_value(data.email),
-            "encrypted_value": {"value": data.email},
-        }
-        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+    # consent requests, unlike privacy requests, only accept 1 identity type- email or phone
+    # if both identity types are provided, use identity type if defined in CONFIG.notifications.notification_service_type, else default to email
+    if data.email and data.phone_number:
+        messaging_method = get_messaging_method(
+            CONFIG.notifications.notification_service_type
+        )
+        if messaging_method == MessagingMethod.EMAIL:
+            identity = _get_or_create_provided_identity(
```

Comment:
a bit of a nit, but i wonder if it'd make things a bit cleaner to move the `get_messaging_method` check and the different branches here _within_ the `_get_or_create_provided_identity` utility function. it may lead to a bit less code, and it also may make the top-level endpoint method here a bit more readable - all the details about getting the `identity` instance would be fully abstracted away from the top-level method, which you've already started to do here (and i think is a good direction to go in generally).

that being said, it's not too hard to follow as is, so not a huge deal. it just feels like some of the conditional branching here could be moved out of the top-level function, and possibly consolidated a bit if done more closely to the identity retrieval.

Reply:
hey, this is a good point, I'll give it a go!